gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/messages/GfxdSystemProcedureMessage.java

```diff
@@ -1062,7 +1066,65 @@ String getSQLStatement(Object[] params) throws StandardException {
             .append(params[2]).append(')').toString();
       }
     },
-    
+
+    setLogLevel {
+
+      @Override
+      boolean allowExecution(Object[] params) {
+        // so allowing log level to be set on all nodes including locator
+        return true;
+      }
+
+      @Override
+      public void processMessage(Object[] params, DistributedMember sender)
+          throws StandardException {
+        try {
+          String logClass = (String) params[0];
+          Level level = org.apache.log4j.Level.toLevel((String) params[1]);
+          SanityManager.DEBUG_PRINT(GfxdConstants.TRACE_SYS_PROCEDURES,
+                  "GfxdSystemProcedureMessage: setting log level for class '" + logClass
+                          + "' to " + level);
+          if (logClass.equals("")) {
```

Comment:
Javadoc of method says that logClass is null for root logger. Better to check for both null and empty here and document accordingly.

Reply:
Fixed the java doc and made the data descriptor as not null.